data/transactions/logic/eval.go

```diff
@@ -4553,6 +4553,10 @@ func opAppLocalPut(cx *EvalContext) error {
 		return err
 	}
 
+	if !cx.allowsLocals(addr, cx.appID) {
```

Comment:
Shouldn't this be guarded with a condition on the version being 9 or higher?

Otherwise I don't think this would return true for pre-v9 programs

Reply:
^ Jason's making a good point, the `cx.allowLocals` was introduced in #5035: in `eval.go`'s usage, under `localsReference`, is guarded by `if cx.version >= sharedResourcesVersion`.

Reply:
Yes, that certainly seems right, yet hard to imagine I don't have a test for old versions.  I'll correct and add.

Reply:
I think it's correct, thought I was worried. `allowsLocal` uses the `cx.available` structure to do it's check - that is allows populated, even for pre v9 programs. It's just that normally it's too permissive, so we only use it for v9 programs.  But here we're using it in reverse. It's permissiveness won't be a problem for old versions, it will be a good thing.
I will confirm that I have a test, but note that the test is any use at all of these opcodes pre v9.

Reply:
ah I see... so the relaxation in locals doesn't really matter pre-v9, but does the constraints from v9.  The new tests seem right.